### DISCUSSION TITLE
UNVT Portal operation

### DISCUSSION THREAD

**Discussion by hfu** [UNopenGIS/7#306] [General]
国土地理院内のゲスト・ネットワーク環境に構築された Raspberry Pi 群である UNVT Portal の運用に関するメモの交換です。

- 現在、UNVT Portal に自在にアクセスできるのは @hfu, @smellman, @yuiseki の3名に限られています。
- JICA 研修で研修員の作業向きに活用いただくことを当座の目標にしています。
- 加えて、Smart Maps Bazaar を支える IPFS ノード群が UNVT Portal 内部に構成されています。

UNVT Portal 運用のノウハウにはオープンに使えるものもあるはずであり、この観点から UNVT Portal の運用に関する情報交換等をオープンにここで行いたいと思っています。

**Comment by hfu** [comment]
# 機材の運用割り当て
現在動いている機材を、とりあえず次のように割り当てたいと思います。

## 管理や実験に共通で使う機材
- j2213

## 研修員に使って頂くことを想定する機材
1. m321
2. m323
3. m327
4. m328
5. m331
6. m332
7. m334
8. m337
9. m342
10. m352
11. m357
12. m358
13. m359

## @yuiseki さんが活用する機材 (Smart Maps Agent / LLM 関係)
- m326

## @hfu が活用する機材 (Smart Maps Bazaar 関係)
- m329 (ほぼ未設定だが HDD を接続できている)
- m330 (大容量の microSDを装備している)
- m345 (HDD を接続している)
- m347 (HDD を接続している)
- その他、UNVT Portal 構築以前に使用開始しているノードが同じネットワークで動作しています。m349 など。

機材の台数に不足があれば、追加投入も可能な状況にありますが、とりあえず現在使用可能になっている機材でネットワークのモニターも含めて作業を進められればと思っています。

**Comment by hfu** [comment]
UNVT Portalがぶら下がっているWiFiの名前はvectortilesです。パスワードは必要な際に現場でお知らせします。

## 研修員が m321 等にログインする方法（案）
1. 研修員のPCのWiFi接続先を、通常使うネットワークからvectortilesに切り替える。
2. 研修員のPCのWindows PowerShellから、ユーザ名portalで、パスワードによってj2213にログインする。
3. 研修員は、j2213からmake m321とするなどして、各人に割り当てられたノードにログインする。ここでは、認証は公開鍵方式で行われる。

いかがでしょう。> @smellman 

## 上記を実現するために藤村が行うべきこと
- [x] j2213 にはパスワードでログイン可能になるようにする。
- [x] j2213 にportal用の秘密鍵を配置する。

**Reply by hfu** [reply]
j2213:/home/portal/.ssh/id_ed25519 をセットアップできました。

**Reply by hfu** [reply]
j2213 にパスワードでログインできるようにしました。具体的には、次の作業をしました。

1. `/etc/ssh/sshd_config`を編集し、`PasswordAuthentication yes`とする。
2. `sudo systemctl restart ssh.service`

**Reply by hfu** [reply]
パスワードは、研修で必要になったタイミングで、`sudo passwd portal`として設定すると良いかな、と思っています。

**Comment by hfu** [comment]
## メモ: Smart Maps Bazaar ノードとして活躍している機材
- m330 (microSD)
- m349 (USB HDD)
- m411 (Raspberry Pi 400) - ただし、地理的に別の場所で、別ネットワークから Smart Maps Bazaar に参加

Smart Maps Bazaar ノードのためのコマンドは、https://github.com/unopengis/smbc に置いてある。

**Comment by smellman** [comment]
> 研修員は、j2213からmake m321とするなどして、各人に割り当てられたノードにログインする。ここでは、認証は公開鍵方式で行われる。

これで入った先のユーザがno passwordでsudoが実行できればOKです。
あとで、割り当てられた自分のノードで試して見ます。

**Reply by hfu** [reply]
ありがとうございます！ j2213 から make m321 あるいは ssh m321 とすることでログインできるように設置しました！

詳しくは、j2213:/home/portal/Makefile をご覧ください。

入った先のユーザがno passwordでsudo実行できるようになっていると思います。

**Comment by hfu** [comment]
## 研修用割り当て
研修員14プラス講師1の15を用意し、次の通りとします。

m321
m323
m327
m328
m331
m332
m334
m337
m342
m352
m357
m358
m359
m326
m330

**Comment by hfu** [comment]
当日の様子は https://discordapp.com/channels/1180387931614085130/1182116163233255454 で記録しています。
